amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -51,72 +41,88 @@ public void testMultiStreamDeserializationSuccess() {
             } catch (final IllegalArgumentException iae) {
                 // expected; ignore
             }
         }
     }
 
+    /**
+     * Test Arns that <b>should not</b> match a valid AWS Kinesis stream Arn.
+     */
     @Test
-    public void testInstanceFromArn() {
-        final Arn arn = toArn(KINESIS_REGION);
-        final StreamIdentifier single = StreamIdentifier.singleStreamInstance(arn.toString());
-        final StreamIdentifier multi = StreamIdentifier.multiStreamInstance(arn.toString());
-
-        assertEquals(single, multi);
-        assertEquals(Optional.empty(), single.streamCreationEpochOptional());
-        assertActualStreamIdentifierExpected(arn, single);
+    public void testMultiStreamByArnWithInvalidStreamArnFail() {
+        final String region = KINESIS_REGION.id();
+        for (final Arn invalidStreamArn : Arrays.asList(
+                streamArn("abc", SERVICE,  region, TEST_ACCOUNT_ID, RESOURCE), // invalid partition
+                streamArn(PARTITION, "dynamodb",  region, TEST_ACCOUNT_ID, RESOURCE), // incorrect service
+                streamArn(PARTITION, SERVICE,  null, TEST_ACCOUNT_ID, RESOURCE), // missing region
+                streamArn(PARTITION, SERVICE,  region, null, RESOURCE), // missing account id
+                streamArn(PARTITION, SERVICE,  region, "123456789", RESOURCE), // account id not 12 digits
+                streamArn(PARTITION, SERVICE,  region, "123456789abc", RESOURCE), // 12char alphanumeric account id
+                streamArn(PARTITION, SERVICE,  region, TEST_ACCOUNT_ID, "table/name"), // incorrect resource type
+                Arn.fromString("arn:aws:dynamodb:us-east-2:123456789012:table/myDynamoDBTable") // valid Arn for incorrect resource
+        )) {
+            try {
+                StreamIdentifier.multiStreamInstance(invalidStreamArn, EPOCH);
+                Assert.fail("Arn " + invalidStreamArn + " should not have created a StreamIdentifier");
+            } catch (final IllegalArgumentException iae) {
+                // expected; ignore
+            }
+        }
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testInstanceWithoutEpochOrArn() {
-        mockStatic(StreamARNUtil.class);
-        when(getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID))
-                .thenReturn(Optional.empty());
-
-        try {
-            StreamIdentifier.singleStreamInstance(DEFAULT_ARN.toString());
-        } finally {
-            verifyStatic(StreamARNUtil.class);
-            getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID);
+    @Test
+    public void testMultiStreamByArnWithInvalidCreationEpochFail() {
+        final Arn streamArn = streamArn(PARTITION, SERVICE,  KINESIS_REGION.id(), TEST_ACCOUNT_ID, RESOURCE);
+        for (final long invalidCreationEpoch : Arrays.asList(-123, 0)) {
+            try {
+                StreamIdentifier.multiStreamInstance(streamArn, invalidCreationEpoch);
```

Comment:
1. KISS: replace `streamArn` w/ `DEFAULT_ARN`.
2. KISS: eliminate the try+catch+Assert and just have two very simple tests:
```
@Test(expected = IllegalArgumentException.class)
public void testNegativeCreationEpoch() {
   SI.multiSI(DEFAULT_ARN, -123);
}

@Test(expected = IllegalArgumentException.class)
public void testZeroCreationEpoch() {
   SI.multiSI(DEFAULT_ARN, 0);
}
```

FYI: this could also be managed by JUnit parameters (https://junit.org/junit4/javadoc/4.12/org/junit/runners/Parameterized.html).